### ISSUE TITLE
find: ‘/bitnami/moodledata/sessions/’: No such file or directory

### ISSUE THREAD

**Issue by 3ace** [bitnami/bitnami-docker-moodle#155]
**Description**

Running Docker Compose using the example `docker-compose.yml` results in `find: ‘/bitnami/moodledata/sessions/’: No such file or directory` error message and container could not start.

I've tried restarting the OS and changing the docker-compose version according to another report at #139 but still not helping.

**Steps to reproduce the issue:**

1. Use the provided `docker-compose.yml` file as described in Readme file.
2. Run Docker Compose
3. The container failed to run

**Describe the results you received:**

```mariadb_1  | mariadb 02:43:00.16
moodle_1   | moodle 02:43:01.24
mariadb_1  | mariadb 02:43:00.17 Welcome to the Bitnami mariadb container
moodle_1   | moodle 02:43:01.24 Welcome to the Bitnami moodle container
moodle_1   | moodle 02:43:01.24 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-moodle
moodle_1   | moodle 02:43:01.24 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-moodle/issues
moodle_1   | moodle 02:43:01.25
moodle_1   | moodle 02:43:01.25 INFO  ==> ** Starting Moodle setup **
mariadb_1  | mariadb 02:43:00.17 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb_1  | mariadb 02:43:00.17 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb_1  | mariadb 02:43:00.17
mariadb_1  | mariadb 02:43:00.18 INFO  ==> ** Starting MariaDB setup **
mariadb_1  | mariadb 02:43:00.21 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1  | mariadb 02:43:00.21 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
mariadb_1  | mariadb 02:43:00.22 INFO  ==> Initializing mariadb database
mariadb_1  | mariadb 02:43:00.24 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb_1  | mariadb 02:43:00.25 INFO  ==> Setting user option
mariadb_1  | mariadb 02:43:00.29 INFO  ==> Using persisted data
mariadb_1  | mariadb 02:43:00.33 INFO  ==> Running mysql_upgrade
mariadb_1  | mariadb 02:43:00.34 INFO  ==> Starting mariadb in background
moodle_1   | moodle 02:43:01.28 DEBUG ==> Validating settings in APACHE_* environment variables
moodle_1   | moodle 02:43:01.39 INFO  ==> Configuring PHP options
moodle_1   | moodle 02:43:01.39 INFO  ==> Setting PHP expose_php option
moodle_1   | moodle 02:43:01.43 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
moodle_1   | moodle 02:43:01.47 DEBUG ==> Validating settings in MOODLE_* environment variables...
moodle_1   | moodle 02:43:01.48 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
moodle_1   | moodle 02:43:01.55 INFO  ==> Restoring persisted Moodle installation
moodle_1   | moodle 02:43:04.53 INFO  ==> Trying to connect to the database server
moodle_1   | moodle 02:43:04.53 DEBUG ==> Getting $CFG->dbtype from Moodle configuration
moodle_1   | moodle 02:43:04.54 DEBUG ==> ^\s*(//\s*)?\$CFG->dbtype\s*=>?([^;,]+)[;,]
moodle_1   | moodle 02:43:04.55 DEBUG ==> Getting $CFG->dbhost from Moodle configuration
moodle_1   | moodle 02:43:04.56 DEBUG ==> ^\s*(//\s*)?\$CFG->dbhost\s*=>?([^;,]+)[;,]
moodle_1   | moodle 02:43:04.57 DEBUG ==> Getting 'dbport' from Moodle configuration
moodle_1   | moodle 02:43:04.57 DEBUG ==> ^\s*(//\s*)?'dbport'\s*=>?([^;,]+)[;,]
moodle_1   | moodle 02:43:04.58 DEBUG ==> Getting $CFG->dbname from Moodle configuration
moodle_1   | moodle 02:43:04.59 DEBUG ==> ^\s*(//\s*)?\$CFG->dbname\s*=>?([^;,]+)[;,]
moodle_1   | moodle 02:43:04.60 DEBUG ==> Getting $CFG->dbuser from Moodle configuration
moodle_1   | moodle 02:43:04.61 DEBUG ==> ^\s*(//\s*)?\$CFG->dbuser\s*=>?([^;,]+)[;,]
moodle_1   | moodle 02:43:04.62 DEBUG ==> Getting $CFG->dbpass from Moodle configuration
moodle_1   | moodle 02:43:04.62 DEBUG ==> ^\s*(//\s*)?\$CFG->dbpass\s*=>?([^;,]+)[;,]
moodle_1   | moodle 02:43:04.63 DEBUG ==> Executing SQL command:
moodle_1   | SELECT 1
moodle_1   | ERROR 2002 (HY000): Can't connect to MySQL server on 'mariadb' (115)
mariadb_1  | mariadb 02:43:07.20 INFO  ==> Stopping mariadb
mariadb_1  |
mariadb_1  | mariadb 02:43:08.21 INFO  ==> ** MariaDB setup finished! **
mariadb_1  | mariadb 02:43:08.25 INFO  ==> ** Starting MariaDB **
mariadb_1  | 2020-11-17  2:43:08 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.5.8-MariaDB) starting as process 1 ...
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Using Linux native AIO
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Uses event mutexes
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Number of pools: 1
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
mariadb_1  | 2020-11-17  2:43:08 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: 128 rollback segments are active.
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Creating shared tablespace for temporary tables
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: 10.5.8 started; log sequence number 20622392; transaction id 81218
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2020-11-17  2:43:08 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1  | 2020-11-17  2:43:08 0 [Note] Server socket created on IP: '0.0.0.0'.
mariadb_1  | 2020-11-17  2:43:08 0 [Warning] 'proxies_priv' entry '@% root@3bdb3ff492ab' ignored in --skip-name-resolve mode.
mariadb_1  | 2020-11-17  2:43:08 0 [Note] Reading of all Master_info entries succeeded
mariadb_1  | 2020-11-17  2:43:08 0 [Note] Added new Master_info '' to hash table
mariadb_1  | 2020-11-17  2:43:08 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
mariadb_1  | Version: '10.5.8-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
mariadb_1  | 2020-11-17  2:43:08 0 [Note] InnoDB: Buffer pool(s) load completed at 201117  2:43:08
moodle_1   | moodle 02:43:09.66 DEBUG ==> Executing SQL command:
moodle_1   | SELECT 1
moodle_1   | 1
moodle_1   | moodle 02:43:09.68 INFO  ==> Running database upgrade
moodle_1   | No upgrade needed for the installed version 3.10 (Build: 20201109) (2020110900). Thanks for coming anyway!
moodle_1   |
moodle_1   | find: ‘/bitnami/moodledata/sessions/’: No such file or directory
moodle_moodle_1 exited with code 1
```

**Describe the results you expected:**

The container should run successfully

**Version**

- Output of `docker version`:

```
Docker version 19.03.13, build 4484c46d9d
```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 2
 Server Version: 19.03.13
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fba4e9a7d01810a393d5d25a3621dc101981175
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 4.15.0-123-generic
 Operating System: Ubuntu 18.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 9.758GiB
 Name: Moodle
 ID: GBPV:W46W:LWII:T7WJ:Q23M:TTPA:FX7N:IQFN:R4BP:5P24:54PS:UCRZ
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.27.4, build 40524192
```

**Comment by andresbono** [comment]
Hi @3ace, thanks for opening this issue.

1) Can you confirm that this is the docker-compose that you are using? https://github.com/bitnami/bitnami-docker-moodle/blob/master/docker-compose.yml

It seems the docker volumes already exist when you deploy the docker-compose:

```
moodle_1   | moodle 02:43:01.55 INFO  ==> Restoring persisted Moodle installation
```

2) Can you delete those volumes and try again? Please make sure your docker-compose deployment doesn't have any relevant data before doing it. If you delete the volumes you will loose that data.

3) If the error is still happening, can you change the entrypoint of the container in the docker-compose and execute:

```
ls -la /bitnami/moodle /bitnami/moodledata
```

**Comment by 3ace** [comment]
Removing the previous volumes indeed enable the container running successfully, but if I restart the container the problem returns.

This is the output of `ls -ls /bitnami/moodle /bitnami/moodledata`

```
moodle_1   | /bitnami/moodle:
moodle_1   | total 1064
moodle_1   | drwxrwxr-x 58 daemon root     4096 Nov 17 18:53 .
moodle_1   | drwxr-xr-x  4 root   root     4096 Nov 17 14:34 ..
moodle_1   | drwxrwxr-x 13 daemon root     4096 Nov 17 18:53 admin
moodle_1   | drwxrwxr-x  5 daemon root     4096 Nov 17 18:53 analytics
moodle_1   | drwxrwxr-x 17 daemon root     4096 Nov 17 18:53 auth
moodle_1   | drwxrwxr-x  6 daemon root     4096 Nov 17 18:53 availability
moodle_1   | -rw-rw-r--  1 daemon root     7380 Nov 17 18:53 babel-plugin-add-module-to-define.js
moodle_1   | drwxrwxr-x  8 daemon root     4096 Nov 17 18:53 backup
moodle_1   | drwxrwxr-x  8 daemon root     4096 Nov 17 18:53 badges
moodle_1   | -rw-rw-r--  1 daemon root      302 Nov 17 18:53 behat.yml.dist
moodle_1   | drwxrwxr-x 47 daemon root     4096 Nov 17 18:53 blocks
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 blog
moodle_1   | -rw-rw-r--  1 daemon root     1162 Nov 17 18:53 brokenfile.php
moodle_1   | -rw-rw-r--  1 daemon root        0 Nov 17 18:53 .buildcomplete
moodle_1   | drwxrwxr-x  6 daemon root     4096 Nov 17 18:53 cache
moodle_1   | drwxrwxr-x  7 daemon root     4096 Nov 17 18:53 calendar
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 cohort
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 comment
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 competency
moodle_1   | drwxrwxr-x  5 daemon root     4096 Nov 17 18:53 completion
moodle_1   | -rw-rw-r--  1 daemon root      552 Nov 17 18:53 composer.json
moodle_1   | -rw-rw-r--  1 daemon root   161252 Nov 17 18:53 composer.lock
moodle_1   | -rw-rw-r--  1 daemon root    56572 Nov 17 18:53 config-dist.php
moodle_1   | -rw-r--r--  1 root   daemon    946 Nov 17 18:53 config.php
moodle_1   | drwxrwxr-x  7 daemon root     4096 Nov 17 18:53 contentbank
moodle_1   | -rw-rw-r--  1 daemon root     2356 Nov 17 18:53 CONTRIBUTING.txt
moodle_1   | -rw-rw-r--  1 daemon root    35147 Nov 17 18:53 COPYING.txt
moodle_1   | drwxrwxr-x 10 daemon root     4096 Nov 17 18:53 course
moodle_1   | drwxrwxr-x  7 daemon root     4096 Nov 17 18:53 customfield
moodle_1   | drwxrwxr-x  8 daemon root     4096 Nov 17 18:53 dataformat
moodle_1   | -rw-rw-r--  1 daemon root     2595 Nov 17 18:53 draftfile.php
moodle_1   | drwxrwxr-x 19 daemon root     4096 Nov 17 18:53 enrol
moodle_1   | drwxrwxr-x  2 daemon root     4096 Nov 17 18:53 error
moodle_1   | -rw-rw-r--  1 daemon root     2404 Nov 17 18:53 .eslintignore
moodle_1   | -rw-rw-r--  1 daemon root     8067 Nov 17 18:53 .eslintrc
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 favourites
moodle_1   | -rw-rw-r--  1 daemon root     3871 Nov 17 18:53 file.php
moodle_1   | drwxrwxr-x  5 daemon root     4096 Nov 17 18:53 files
moodle_1   | drwxrwxr-x 18 daemon root     4096 Nov 17 18:53 filter
moodle_1   | -rw-rw-r--  1 daemon root      597 Nov 17 18:53 .gherkin-lintrc
moodle_1   | -rw-rw-r--  1 daemon root       94 Nov 17 18:53 .gitattributes
moodle_1   | drwxrwxr-x  2 daemon root     4096 Nov 17 18:53 .github
moodle_1   | -rw-rw-r--  1 daemon root     1103 Nov 17 18:53 .gitignore
moodle_1   | drwxrwxr-x 11 daemon root     4096 Nov 17 18:53 grade
moodle_1   | drwxrwxr-x  5 daemon root     4096 Nov 17 18:53 group
moodle_1   | -rw-rw-r--  1 daemon root     6725 Nov 17 18:53 GruntfileComponents.js
moodle_1   | -rw-rw-r--  1 daemon root    33641 Nov 17 18:53 Gruntfile.js
moodle_1   | drwxrwxr-x  9 daemon root     4096 Nov 17 18:53 h5p
moodle_1   | -rw-rw-r--  1 daemon root     1423 Nov 17 18:53 help_ajax.php
moodle_1   | -rw-rw-r--  1 daemon root     1872 Nov 17 18:53 help.php
moodle_1   | -rw-rw-r--  1 daemon root     4717 Nov 17 18:53 index.php
moodle_1   | drwxrwxr-x  3 daemon root     4096 Nov 17 18:53 install
moodle_1   | -rw-rw-r--  1 daemon root    26188 Nov 17 18:53 install.php
moodle_1   | -rw-rw-r--  1 daemon root      664 Nov 17 18:53 INSTALL.txt
moodle_1   | drwxrwxr-x  3 daemon root     4096 Nov 17 18:53 iplookup
moodle_1   | -rw-rw-r--  1 daemon root       16 Nov 17 18:53 .jshintignore
moodle_1   | -rw-rw-r--  1 daemon root     1545 Nov 17 18:53 .jshintrc
moodle_1   | drwxrwxr-x  3 daemon root     4096 Nov 17 18:53 lang
moodle_1   | drwxrwxr-x 65 daemon root     4096 Nov 17 18:53 lib
moodle_1   | drwxrwxr-x  2 daemon root     4096 Nov 17 18:53 licenses
moodle_1   | drwxrwxr-x  2 daemon root     4096 Nov 17 18:53 local
moodle_1   | drwxrwxr-x  3 daemon root     4096 Nov 17 18:53 login
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 media
moodle_1   | drwxrwxr-x  7 daemon root     4096 Nov 17 18:53 message
moodle_1   | drwxrwxr-x  6 daemon root     4096 Nov 17 18:53 mnet
moodle_1   | drwxrwxr-x 25 daemon root     4096 Nov 17 18:53 mod
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 my
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 notes
moodle_1   | -rw-rw-r--  1 daemon root   354972 Nov 17 18:53 npm-shrinkwrap.json
moodle_1   | -rw-rw-r--  1 daemon root        9 Nov 17 18:53 .nvmrc
moodle_1   | -rw-rw-r--  1 daemon root     1360 Nov 17 18:53 package.json
moodle_1   | drwxrwxr-x  7 daemon root     4096 Nov 17 18:53 payment
moodle_1   | -rw-rw-r--  1 daemon root    10057 Nov 17 18:53 phpunit.xml.dist
moodle_1   | drwxrwxr-x 14 daemon root     4096 Nov 17 18:53 pix
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 plagiarism
moodle_1   | -rw-rw-r--  1 daemon root     1677 Nov 17 18:53 pluginfile.php
moodle_1   | drwxrwxr-x 10 daemon root     4096 Nov 17 18:53 portfolio
moodle_1   | drwxrwxr-x  6 daemon root     4096 Nov 17 18:53 privacy
moodle_1   | -rw-rw-r--  1 daemon root      266 Nov 17 18:53 PULL_REQUEST_TEMPLATE.txt
moodle_1   | drwxrwxr-x 11 daemon root     4096 Nov 17 18:53 question
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 rating
moodle_1   | -rw-rw-r--  1 daemon root     1176 Nov 17 18:53 README.txt
moodle_1   | drwxrwxr-x 21 daemon root     4096 Nov 17 18:53 report
moodle_1   | drwxrwxr-x 28 daemon root     4096 Nov 17 18:53 repository
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 rss
moodle_1   | drwxrwxr-x  8 daemon root     4096 Nov 17 18:53 search
moodle_1   | -rw-rw-r--  1 daemon root       67 Nov 17 18:53 .shifter.json
moodle_1   | -rw-rw-r--  1 daemon root     2446 Nov 17 18:53 .stylelintignore
moodle_1   | -rw-rw-r--  1 daemon root     4727 Nov 17 18:53 .stylelintrc
moodle_1   | drwxrwxr-x  5 daemon root     4096 Nov 17 18:53 tag
moodle_1   | drwxrwxr-x  4 daemon root     4096 Nov 17 18:53 theme
moodle_1   | -rw-rw-r--  1 daemon root     1600 Nov 17 18:53 tokenpluginfile.php
moodle_1   | -rw-rw-r--  1 daemon root     2191 Nov 17 18:53 TRADEMARK.txt
moodle_1   | -rw-rw-r--  1 daemon root    10048 Nov 17 18:53 .travis.yml
moodle_1   | drwxrwxr-x  9 daemon root     4096 Nov 17 18:53 user
moodle_1   | drwxrwxr-x  2 daemon root     4096 Nov 17 18:53 userpix
moodle_1   | -rw-rw-r--  1 daemon root     1634 Nov 17 18:53 version.php
moodle_1   | drwxrwxr-x  7 daemon root     4096 Nov 17 18:53 webservice
moodle_1   |
moodle_1   | /bitnami/moodledata:
moodle_1   | total 36
moodle_1   | drwxrwxr-x 8 daemon root   4096 Nov 17 18:55 .
moodle_1   | drwxr-xr-x 4 root   root   4096 Nov 17 14:34 ..
moodle_1   | drwxrwxr-x 4 daemon daemon 4096 Nov 17 18:54 cache
moodle_1   | drwxrwxr-x 7 daemon daemon 4096 Nov 17 18:51 filedir
moodle_1   | -rw-rw-rw- 1 daemon daemon  128 Nov 17 18:49 .htaccess
moodle_1   | drwxrwxr-x 2 daemon daemon 4096 Nov 17 18:49 lang
moodle_1   | drwxrwxr-x 5 daemon daemon 4096 Nov 17 18:54 localcache
moodle_1   | -rw-r--r-- 1 daemon daemon    0 Nov 17 18:54 moodle-cron.log
moodle_1   | drwxrwxr-x 2 daemon daemon 4096 Nov 17 18:53 muc
moodle_1   | drwxrwxr-x 5 daemon daemon 4096 Nov 17 18:52 temp
```

**Comment by andresbono** [comment]
It seems the sessions directory doesn't exist until the web application is accessed for the first time:

```console
$ docker pull docker.io/bitnami/moodle:3-debian-10
$ docker-compose up -d # Wait until the application is initialized
$ docker-compose exec moodle ls /bitnami/moodledata/sessions
ls: cannot access '/bitnami/moodledata/sessions': No such file or directory
$ docker-compose exec moodle curl localhost:8080
$ docker-compose exec moodle ls /bitnami/moodledata/sessions
sess_3n1sstq0vls459rm08liedrsd5
```

In the block of code below, we can modify the condition to only execute the find command if the `/bitnami/moodledata/sessions` directory exists:

https://github.com/bitnami/bitnami-docker-moodle/blob/28fb0a20efc079dad9f093837c9198bc50973208/3/debian-10/rootfs/opt/bitnami/scripts/libmoodle.sh#L168-L173

We will release a new version of the container and we will let you know.

**Comment by andresbono** [comment]
Hi @3ace, we released a new version of the container: [3.10.0-debian-10-r12](https://github.com/bitnami/bitnami-docker-moodle/commit/275e5de7500fb6f19689e581c76c8d482c2dbc3b). Could you please give it a try and let us know if the issue is solved?

**Comment by 3ace** [comment]
@andresbono looks like it fixed the problem. I've tested running and restarting the container.

Thanks a lot
